test/preview/components/file_converters/test_azure_ocr_doc_converter.py

```diff
@@ -0,0 +1,100 @@
+
+
+class TestAzureOCRDocumentConverter:
+    @pytest.mark.unit
+    def test_to_dict(self):
+        component = AzureOCRDocumentConverter(endpoint="test_endpoint", credential_key="test_credential_key")
+        data = component.to_dict()
+        assert data == {
+            "type": "AzureOCRDocumentConverter",
+            "init_parameters": {
+                "credential_key": "test_credential_key",
+                "endpoint": "test_endpoint",
+                "id_hash_keys": [],
+                "model_id": "prebuilt-read",
+                "save_json": False,
+            },
+        }
+
+    @pytest.mark.unit
+    def test_from_dict(self):
+        data = {
+            "type": "AzureOCRDocumentConverter",
+            "init_parameters": {
+                "credential_key": "test_credential_key",
+                "endpoint": "test_endpoint",
+                "id_hash_keys": [],
+                "model_id": "prebuilt-read",
+                "save_json": False,
+            },
+        }
+        component = AzureOCRDocumentConverter.from_dict(data)
+        assert component.endpoint == "test_endpoint"
+        assert component.credential_key == "test_credential_key"
+        assert component.id_hash_keys == []
+        assert component.model_id == "prebuilt-read"
+        assert component.save_json is False
+
+    @pytest.mark.unit
+    def test_run(self, preview_samples_path):
+        with patch("haystack.preview.components.file_converters.azure.DocumentAnalysisClient") as mock_azure_client:
+            mock_result = Mock(pages=[Mock(lines=[Mock(content="mocked line 1"), Mock(content="mocked line 2")])])
+            mock_azure_client.return_value.begin_analyze_document.return_value.result.return_value = mock_result
+
+            component = AzureOCRDocumentConverter(endpoint="test_endpoint", credential_key="test_credential_key")
+            output = component.run(paths=[preview_samples_path / "pdf" / "sample_pdf_1.pdf"])
+            document = output["documents"][0]
+            assert document.text == "mocked line 1\nmocked line 2\n\f"
+
+    @pytest.mark.integration
+    @pytest.mark.skipif(
+        "AZURE_FORMRECOGNIZER_ENDPOINT" not in os.environ and "AZURE_FORMRECOGNIZER_KEY" not in os.environ,
+        reason="Azure credentials not available",
+    )
+    def test_run_with_pdf_file(self, preview_samples_path):
+        component = AzureOCRDocumentConverter(
+            endpoint=os.environ["AZURE_FORMRECOGNIZER_ENDPOINT"], credential_key=os.environ["AZURE_FORMRECOGNIZER_KEY"]
+        )
+        output = component.run(paths=[preview_samples_path / "pdf" / "sample_pdf_1.pdf"])
+        documents = output["documents"]
+        assert len(documents) == 1
+        assert "A sample PDF file" in documents[0].text
+        assert "Page 2 of Sample PDF" in documents[0].text
+        # Azure free tier limits the extraction to 2 pages
+        # assert "Page 4 of Sample PDF" in documents[0].text
```

Comment:
Wow, that's not very generous of them :sweat_smile: Maybe we should consider adding other providers for OCR in the near future.

Reply:
Well, I guess 2 pages is enough for a user to decide whether the tool is good enough them or not to pay for it :D But I agree, having an OCR component that's more generous would be nice.